sshd/src/test/scala/ammonite/repl/sshd/SshServerTests.scala

```diff
@@ -0,0 +1,89 @@
+package ammonite.repl.sshd
+
+import ammonite.ops.Path
+import com.jcraft.jsch.{JSchException, Session}
+import org.scalacheck.Prop.forAll
+import utest._
+import utest.framework.TestSuite
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Promise}
+import scala.language.postfixOps
+
+object SshServerTests extends TestSuite with ScalaCheckSupport with SshTestingTraits {
+  override val tests = TestSuite {
+    'canConnectAndAuthenticate{
+      withTmpDirectory { implicit tmpDir ⇒
```

Comment:
`withTmpDirectory` can probably be combined with `withTestSshServer`; there are no cases where one is called but not the other AFAICT, so we can avoid the boilerplate around every test case.